app/routes.py

```diff
@@ -1,2 +1,239 @@
-from flask import Blueprint
+from flask import Blueprint, json, jsonify, request, make_response
+from werkzeug.wrappers import response
+from app.models.task import Task
+from app.models.goal import Goal
+from app import db
+import requests
+from datetime import datetime
+import os
+from dotenv import load_dotenv
+
+tasks_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+
+@tasks_bp.route("", methods=["POST"])
+def post_tasks():
+    request_body = request.get_json()
+    if "title" not in request_body or "description" not in request_body or "completed_at" not in request_body:
+        return make_response({"details":"Invalid data"}, 400)
+    
+    new_task = Task(
+        title = request_body["title"],
+        description = request_body["description"],
+        completed_at = request_body["completed_at"]
+        )
+    db.session.add(new_task)
+    db.session.commit()
+    
+    response_body = {"task": {"id": new_task.id,
+        "title": new_task.title,
+        "description": new_task.description,
+        "is_complete": bool(new_task.completed_at)}}
+        
+    return jsonify(response_body), 201
+
+
+@tasks_bp.route("", methods=["GET"])   
+def get_tasks():
+        sort_query = request.args.get("sort")
+        if sort_query == "asc":
+            tasks = Task.query.order_by(Task.title.asc()).all()
+        elif sort_query == "desc":
+            tasks = Task.query.order_by(Task.title.desc()).all()
+        else:
+            tasks = Task.query.all()
+
+        tasks_response = []
+        for task in tasks: 
+            tasks_response.append({
+                "id": task.id,
+                "title": task.title,
+                "description": task.description,
+                "is_complete": bool(task.completed_at)
+                })
+
+        return jsonify(tasks_response), 200
+
+
+@tasks_bp.route("/<task_id>", methods=["GET", "PUT", "DELETE"])
+def handle_task(task_id):
+    task = Task.query.get_or_404(task_id)
+    if not task:
+        return make_response(f"Task {task_id} not found", 404)
```

Comment:
Note that calling `make_response` on a string doesn't return a JSON response. Instead it returns an html response. Since we're building a JSON-based web api, we should make sure that we are consistently returning JSON (by using jsonify here).